shared/src/commonMain/kotlin/com/vml/tutorial/plantshop/plants/data/PlantsRepository.kt

```diff
@@ -2,15 +2,41 @@ package com.vml.tutorial.plantshop.plants.data
 
 import com.vml.tutorial.plantshop.plants.domain.Plant
 import com.vml.tutorial.plantshop.plants.domain.PlantsDataSource
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.IO
+import kotlinx.coroutines.SupervisorJob
+import kotlinx.coroutines.launch
 
 interface PlantsRepository {
     fun getPlants(): List<Plant>
+    fun getFavorites(): List<Plant>
+    fun toggleFavoriteStatus(plant: Plant)
 }
 
 class PlantsRepositoryImpl(
     private val localDataSource: PlantsDataSource
-): PlantsRepository {
+) : PlantsRepository {
     override fun getPlants(): List<Plant> {
         return localDataSource.getPlants()
     }
-}
\ No newline at end of file
+
+    override fun getFavorites(): List<Plant> {
```

Comment:
I think you can create a state flow here (PlantsRepository) to make it observable on other screens. So if it changes, the new fav state will be emitted on the particular screen.

Reply:
State flow is created